types/user.ts

```diff
@@ -247,6 +248,51 @@ export class User extends LazyObject {
     ]);
   }
 
+  /** getReportsCursor
+  * get a QueryCursor for a Comment's reports starting from most recent
+  */
+  public getReportsCursor() {
+    return new QueryCursor(
+      Report,
+      5,
+      collection(db, 'reports'),
+      where('reporter', '==', this.docRef!),
+      orderBy('timestamp', 'desc')
+    );
+  }
+
+  /** clearEffects
+   * delete all reports, posts, and comments by this user
+   *
+   */
+  public async clearEffects() {
+    if (!this.docRef) return;
+    const tasks: any[] = [];
+
+    // delete all their reports
+    console.log('about to delete reports from user');
+
+    (
+    await this.getReportsCursor().________getAll_CLOWNTOWN_LOTS_OF_READS()
```

Comment:
lol okay maybe we should rename the function getAll, I can see it has actual non-clowntown uses :p

Reply:
Separate PR tho